src/components/WeeklyRoutinePage/WeekDayCard.js

```diff
@@ -86,7 +81,7 @@ export default function WeekDayCard({ name, data, products }) {
               {mornings.map((morning) => {
                 return (
                   <li className="product-name-left">
-                    {getProductsById(morning.id)}
+                    {getProductsById(morning.id, products).name}
```

Comment:
Here you could do what I said in a few comments before. Just get it once and store in a variable and use it later instead of calling the function every time

Reply:
I am calling the function with different arguments (first with morning.id, second with evening.id), so i will not get around calling it twice. Or is there a trick, I am missing?

Reply:
ah okay,, now I see. Sorry I thought it was the same as before.